rxnetty/src/main/java/io/reactivex/netty/client/ConnectionPoolImpl.java

```diff
@@ -169,23 +198,27 @@ public void call(final Subscriber<? super ObservableConnection<I, O>> subscriber
         }
 
         long startTimeMillis = Clock.newStartTimeMillis();
+        aquiredConnectionsCounter.decrementAndGet();
 
         try {
             connection.getChannel().pipeline().fireUserEventTriggered(new PooledConnectionReleasedEvent(connection));
             metricEventsSubject.onEvent(ClientMetricsEvent.POOL_RELEASE_START);
-            if (isShutdown.get() || !connection.isUsable()) {
+            if (isShutdownRequested.get() || !connection.isUsable()) {
                 discardConnection(connection);
                 metricEventsSubject.onEvent(ClientMetricsEvent.POOL_RELEASE_SUCCESS, Clock.onEndMillis(
                         startTimeMillis));
+                performShutdownIfRequested();
```

Comment:
I think it will be better to call this when `aquiredConnectionsCounter.decrementAndGet();` above returns `0`